test/tools/router_check/router.cc

```diff
@@ -0,0 +1,211 @@
+#include "test/tools/router_check/router.h"
+
+Json::ObjectPtr RouterCheckTool::loadJson(const std::string& config_json,
+                                          const std::string& schema) {
+
+  try {
+    // Json configuration
+    Json::ObjectPtr loader = Json::Factory::LoadFromFile(config_json);
+    loader->validateSchema(schema);
+    return loader;
+
+  } catch (const EnvoyException& ex) {
+    std::cerr << "config schema JSON load failed: " << config_json << std::endl;
+    std::cerr << ex.what() << std::endl;
```

Comment:
Whether or not to throw exceptions seems to be philosophical. In this case, I've chosen to print to std::err. The tool then exits main with return 1.